### PR TITLE
Schema tracking global routing

### LINKED ISSUE #16516: Feature Request: Global routing doesn't work with tables not in vschema despite schema tracking
### Feature Description

If there are some tables that aren't specified in the vschema (This can be the case especially for unsharded keyspaces), then global routing doesn't work for them despite schema tracking running.

It would be nice if we could get the following scenario to work - 
1. A user sets up 2 keyspaces, one of which is unsharded.
2. They create a table in the unsharded keyspace but don't add it to the Vschema.
3. If schema tracking is running, if the table created has a globally unique name, then we should support global routing on it.

### Use Case(s)

Described ☝️

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR implements the feature requested in #16516.
The fix is pretty straight-forward. Since schema tracking can add tables to the vschema that were previously not there, we should recompute the global tables map that we use for global routing.

This PR also adds an e2e test that verifies the use case described in the issue now works.

## Related Issue(s)

- Fixes #16516

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtgate/vindexes/vschema.go [hunk 1]
```diff
@@ -331,9 +331,9 @@ func BuildVSchema(source *vschemapb.SrvVSchema, parser *sqlparser.Parser) (vsche
 		created:        time.Now(),
 	}
 	buildKeyspaces(source, vschema, parser)
-	// buildGlobalTables before buildReferences so that buildReferences can
+	// BuildGlobalTables before buildReferences so that buildReferences can
 	// resolve sources which reference global tables.
-	buildGlobalTables(source, vschema)
+	BuildGlobalTables(source, vschema)
 	buildReferences(source, vschema)
```
Comment (harshit-gangal) [inline go/vt/vtgate/vindexes/vschema.go:337]: in this sequence of operations, we remove auto increment table i.e. sequence tables from global tables on error.
The second call can add it back again.

look at `resolveAutoIncrement`

**Review by harshit-gangal — COMMENTED** [review]
I think we might mark almost all tables as ambiguous if we call the build global tables, again.

Could you verify with a test?

**Review by systay — APPROVED** [review]

**Comment by GuptaManan100** [comment]
@harshit-gangal Yes verified that. That's when we build the global table again, we create a new map

File: go/vt/vtgate/vindexes/vschema.go (hunk 1 above)
Comment (GuptaManan100) [reply go/vt/vtgate/vindexes/vschema.go:337]: But this problem exists with schema tracking too, right? It would add the table back to the vschema.

Comment (harshit-gangal) [reply go/vt/vtgate/vindexes/vschema.go:337]: In a way it is not a problem, as those tables added does not have the column vindex information, so no routing can happen on them.

**Review by harshit-gangal — APPROVED** [review]

**Comment by GuptaManan100** [comment]
I looked at this today and I see what's happening. In MoveTables, the user is trying to move a table from one keyspace to another. As part of this, MoveTables creates the vschema for new keyspace and adds the table there. Also, when MoveTables completes it removes the table from the vschema for the source keyspace. This means that if the table was globally routable before, it continues to be globally routable after move tables since the table exists only in 1 Vschema. What's happening after the change in the PR is that because the table still exists in the source (and not dropped), schema tracker populatest the table information back into the vschema! And then when it tries to build the global tables again, it finds the table with the same name in both the source and the target! This fails the global routing with the error we see above!

I don't know how to fix this.

**Comment by GuptaManan100** [comment]
Given this issue with MoveTables, we have decided to not go ahead with this fix.

**Pull request closed without merging**